test/extended/openstack/loadbalancers.go

```diff
@@ -419,3 +478,56 @@ func createTestDeployment(depName string, labels map[string]string, replicas int
 	}}
 	return testDeployment
 }
+
+// GetFloatingNetworkID returns a floating network ID.
+func GetFloatingNetworkID(client *gophercloud.ServiceClient) (string, error) {
+	type NetworkWithExternalExt struct {
+		networks.Network
+		external.NetworkExternalExt
+	}
+	var allNetworks []NetworkWithExternalExt
+
+	page, err := networks.List(client, networks.ListOpts{}).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	err = networks.ExtractNetworksInto(page, &allNetworks)
+	if err != nil {
+		return "", err
+	}
+
+	for _, network := range allNetworks {
+		if network.External && len(network.Subnets) > 0 {
+			page, err := subnets.List(client, subnets.ListOpts{NetworkID: network.ID}).AllPages()
+			if err != nil {
+				return "", err
+			}
+			subnetList, err := subnets.ExtractSubnets(page)
+			if err != nil {
+				return "", err
+			}
+			for _, networkSubnet := range network.Subnets {
+				subnet := getSubnet(networkSubnet, subnetList)
+				if subnet != nil {
+					if subnet.IPVersion == 4 {
+						return network.ID, nil
+					}
+				} else {
+					return network.ID, nil
```

Comment:
You still wants to return if subnet is not accessible?

Reply:
Yes, in fact that is the typical case: The external network is created by the openstack admin with `--no-share true` so the tenant cannot inspect the external subnets. Please refer to https://github.com/kubernetes/cloud-provider-openstack/pull/2014

Reply:
Alright!